vars/main.yml

```diff
@@ -9,9 +9,9 @@ logging_config_path: /etc/google-fluentd/google-fluentd.conf
 logging_plugins_path: /etc/google-fluentd/plugin/
 logging_validation_cmd: '/usr/sbin/google-fluentd -c %s --dry-run'
 
-ops-agent_service_name: google-cloud-ops-agent.target
+ops_agent_service_name: google-cloud-ops-agent
 ops-agent_config_path: /etc/google-cloud-ops-agent/config.yaml
-ops-agent_validation_cmd: '/opt/google-cloud-ops-agent/subagents/collectd/sbin/collectd -tC %s'
+ops-agent_validation_cmd: '/opt/google-cloud-ops-agent/subagents/fluent-bit/bin/fluent-bit -c %s'
```

Comment:
As mentioned on the issue, this does not validate the ops agent config. (Nor did the old one). `collectd` and `fluent-bit` do not directly read the ops agent config.

Also, we are about to rename `/opt/google-cloud-ops-agent`, and you should not depend on anything contained inside.

Reply:
After inspecting the systemd service file for the agent, I have switched to:
```
/opt/google-cloud-ops-agent/libexec/google_cloud_ops_agent_engine -in /etc/google-cloud-ops-agent/config.yaml
```

Reply:
The name of the file gets substituted for "%s" which [needs to be present in the validation command](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/copy_module.html#parameter-validate). 
So this could just be `/opt/google-cloud-ops-agent/libexec/google_cloud_ops_agent_engine -in %s`. 

Does this work for v 1.*.* as well? We could also consider just making this a no-op `echo %s > /dev/null`.

Reply:
Again, this is not part of the API for Ops Agent and it *will* break soon in a minor patch release. We are planning to move those files soon. We can talk about adding an intentional API for config validation, but there is not one now.

`/bin/true` takes arguments for this very kind of use case, so you don't need an `echo` just to redirect it to `/dev/null`. (That said, I think it's poor form to do unnecessary work in an Ansible playbook.)

Reply:
I agree that this is not quite the right thing:

1.  It should use `%s` as @rmoriar1 mentions. This isn't just nice. The file gets copied to a temporary location before being validated. It is only copied to the correct destination if it passes.
2. It's not really doing validation. It's creating honest to goodness configuration files.
3. The file location issue noted by @quentinmit 

I'd get rid of validation since this step is performed in the systemd unit file anyhow. But, if you want it, I think this is better:

`{{ some_prefix_fact_that_will_get_compilcated}}/google_cloud_ops_agent_engine -in %s -out /dev/null`